client/lib/user/user.js

```diff
@@ -242,9 +242,12 @@ User.prototype.changeUser = function( username, password, callback ) {
 	if ( config.isEnabled( 'support-user' ) ) {
 		wpcom.changeUser( username, password, function( error ) {
 			if ( ! error ) {
+				this.once( 'change', () => callback() );
+				this.clear();
 				this.fetch();
+			} else {
+				callback( error );
 			}
-			callback( error );
```

Comment:
how about instead of merging this into a nested structure we just abort early

```js
if ( error ) {
	return callback( error );
}

this.once( 'change', callback ); // no need to create a closure for calling callback
this.clear();
this.fetch();
```